### PR TITLE
Sliu40

### PR DESCRIPTION
**Pull request by voice-liu**
image

### CODE REVIEW

**Comment by PohotonicCrystal** [comment]
The UML need to update since you implement service and interface but not show on UML

**Comment by caroline666-t** [comment]
> image

1. Standardize the capitalization for properties
2. You implement service and interface but not show on UML

File: TodoItems.Core/BizException.cs [hunk 1]
```diff
@@ -0,0 +1,6 @@
+﻿namespace TodoItems.Core.BizException
```
Comment (caroline666-t) [inline TodoItems.Core/BizException.cs:1]: What is Biz?

File: TodoItems.Core/Constants.cs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+namespace TodoItems.Core;
+
+public static class Constants
+{
+    public const int MAX_DAY_SAME_DUEDAY = 8;
+    public const int MAX_MODIFY_TIME_ONE_DAY = 3;
+
```
Comment (caroline666-t) [inline TodoItems.Core/Constants.cs:7]: Good job!

File: TodoItems.Core/Model/Modification.cs [hunk 3]
```diff
@@ -0,0 +1,14 @@
+﻿namespace TodoItems.Core.Model
+{
+    public class Modification
+    {
+        //public string NewDescription { get; set; }
+        //public string OldDescription { get; set; }
```
Comment (caroline666-t) [inline TodoItems.Core/Model/Modification.cs:6]: Remove useless comments

File: TodoItems.Core/Constants.cs [hunk 4]
```diff
@@ -0,0 +1,9 @@
+namespace TodoItems.Core;
+
+public static class Constants
```
Comment (PohotonicCrystal) [inline TodoItems.Core/Constants.cs:3]: good practice.  can add to our code convention

File: TodoItems.Core/Model/TodoItem.cs [hunk 5]
```diff
@@ -0,0 +1,43 @@
+﻿using System.Security.Cryptography;
+using TodoItems.Core.BizException;
+namespace TodoItems.Core.Model;
+
+public class TodoItem
+{
+    public string _id;
+    public string Description { get; set; }
+    public DateOnly DueDay { get; set; }
```
Comment (caroline666-t) [inline TodoItems.Core/Model/TodoItem.cs:9]: Amazing use of dateonly!

File: TodoItems.Core/Model/TodoItem.cs [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    public void Modify(string description)
+    {
+        int count = 0;
+        ModificationList.ForEach(modification =>
+        {
+            if (DateTime.Now.Subtract(modification.TimeStamp).TotalDays <= 1)
+            {
+                count++;
+            }
+        });
```
Comment (caroline666-t) [inline TodoItems.Core/Model/TodoItem.cs:31]: You may use count in linq to simplfy this.

File: TodoItems.Core/Model/TodoItem.cs [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+        if (count >= Constants.MAX_MODIFY_TIME_ONE_DAY)
+        {
+            throw new MaximumModificationException("You have reached the maximum number of modifications for today. Please try agian tomorrow.");
```
Comment (caroline666-t) [inline TodoItems.Core/Model/TodoItem.cs:34]: I recommend to write this message in your exception class.

File: TodoItems.Core/Service/TodoItemService.cs [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+        public TodoItemService(ITodosRepository repository)
+        {
+            _repository = repository;
+        }
+
+        public TodoItem Create(string description, DateOnly dueDay, string userId)
+        {
+            if(dueDay < DateOnly.FromDateTime(DateTime.Now))
+            {
+                throw new DueDayEarlyException($"canot ealier than today");
```
Comment (caroline666-t) [inline TodoItems.Core/Service/TodoItemService.cs:20]: canot should be cannot

Comment (anyan-0110) [inline TodoItems.Core/Service/TodoItemService.cs:20]: small type, "cannot"

File: TodoItems.Core/Repository/ITodosRepository.cs [hunk 9]
```diff
@@ -0,0 +1,10 @@
+﻿using TodoItems.Core.Model;
+
+namespace TodoItems.Core.Repository
+{
+    public interface ITodosRepository
+    {
+        List<TodoItem> FindAllTodoItemsByUserIdAndDueDay(string userId,DateOnly dueDay);
```
Comment (PohotonicCrystal) [inline TodoItems.Core/Repository/ITodosRepository.cs:7]: can return a number

File: TodoItems.Test/TodoItemServiceTest.cs [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+
+        _service = new TodoItemService(_mockedTodosRepository.Object);
+
+        var todoItem = _service.Create("Des", DateOnly.FromDateTime(DateTime.Now), "user1");
+
+        Assert.NotNull(todoItem);
+    }
+
+    [Fact]
+    public void Can_create_items_10_dueDay()
```
Comment (PohotonicCrystal) [inline TodoItems.Test/TodoItemServiceTest.cs:41]: testMehod name is not consistent with implement

File: TodoItems.Core/Service/TodoItemService.cs [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+
+            List<TodoItem> items= _repository.FindAllTodoItemsByUserIdAndDueDay(userId, dueDay);
+                
+            if (items.Count >= Constants.MAX_DAY_SAME_DUEDAY) 
+            {
+                throw new MaximumSameDueDayException($"too many items in same day for {userId}");
+            }
+
+            TodoItem todoItem = new TodoItem(description, dueDay, userId); ;
+            _repository.Insert(todoItem);
```
Comment (caroline666-t) [inline TodoItems.Core/Service/TodoItemService.cs:31]: Not sure if Insert should implement in interface, to be discussed

File: TodoItems.Test/TodoItemServiceTest.cs [hunk 12]
```diff
@@ -0,0 +1,117 @@
...
+
+namespace TodoItems.Test;
+public class TodoItemServiceTest
+{
+    private TodoItemService _service;
+    private readonly Mock<ITodosRepository> _mockedTodosRepository;
+
+    public TodoItemServiceTest() {
+        _mockedTodosRepository= new Mock<ITodosRepository>();
+        _service = new TodoItemService(_mockedTodosRepository.Object);
```
Comment (caroline666-t) [inline TodoItems.Test/TodoItemServiceTest.cs:15]: Why new service here???

**Pull request closed without merging**